### PR TITLE
refactor: Cleanup old code, fix TS errors

### PR DESCRIPTION
**Pull request by bhelx**
Should have no effect on functionality, pure refactor:

* Cleans up unused code to make it easier to know what is being used and what isn't
* Removes any TS-ignores, tries to use proper types over any
* Removes unused styles

We can wait until Joe gets back to look at this.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -12,57 +11,25 @@ import ModuleLoader from './components/Module-Loader/ModuleLoader';
 import PluginInput from './components/PluginInput/PluginInput';
 import PluginOutput from './components/PluginOutput/PluginOutput';
 import URLInput from './components/URLInput/URLInput';
-import { MimeTypes } from './util/MimeTypes';
-type ModuleUrl = string;
-type ModuleBytes = Uint8Array;
-
-type ModuleData = ModuleUrl | ModuleBytes;
-interface PluginState {
```
Comment (bhelx) [inline src/App.tsx:20]: Moved this stuff to types namespace since it is needed a few places

File: src/App.tsx [hunk 2]
```diff
@@ -12,57 +11,25 @@ import ModuleLoader from './components/Module-Loader/ModuleLoader';
...
-      return { ...state, ...action.payload };
-    case 'UPLOAD_TYPE':
-      return { ...state, uploadType: action.payload };
     case 'URL_INPUT':
-      return { ...state, ...action.payload };
     case 'MODULE_INPUT':
-      return { ...state, ...action.payload };
     case 'FUNCTION_DROPDOWN':
-      return { ...state, ...action.payload };
     case 'FILE_DROP':
```
Comment (bhelx) [inline src/App.tsx:23]: I think some of these are unused now and some we can probably consolidate

File: src/components/Module-Loader/ModuleLoader.tsx [hunk 3]
```diff
@@ -10,26 +9,17 @@ const ModuleLoader: React.FC<Props> = function ({ onChange, moduleName }) {
...
-    let value: Uint8Array;
     let ele = document.getElementById('selected_file') as HTMLInputElement;
     if (ele.files) {
       let file = ele.files[0];
       const moduleName = file.name;
       file
         .arrayBuffer()
-        .then((buffer) => {
-          const UINT_8 = new Uint8Array(buffer);
-          value = UINT_8;
```
Comment (bhelx) [inline src/components/Module-Loader/ModuleLoader.tsx:23]: not sure why we were setting this higher scoped value. it didn't seem needed

File: src/components/PluginInput/PluginInput.tsx [hunk 4]
```diff
@@ -1,32 +1,21 @@
...
   input: Uint8Array;
-  label: string;
-  dropDownTitle: string;
   mimeType: string;
-  dispatch: (action: { type: string; payload: any }) => void;
-  handleOnRun: any;
-  onChange: any;
-  onKeyDown: any;
-  handleDrop: any;
+  dispatch: DispatchFunc;
```
Comment (bhelx) [inline src/components/PluginInput/PluginInput.tsx:8]: Figured it was best to give this type a name. Would be ideal to typecheck  the payload but we don't want to go down the redux rabbithole

File: src/components/PluginInput/PluginInput.tsx [hunk 5]
```diff
@@ -48,6 +37,9 @@ const PluginInput: React.FC<PluginInputProps> = function ({
   const onDrop = async (e: React.DragEvent) => {
     e.preventDefault();
     const files = e.dataTransfer.files;
+    if (files.length > 1) {
```
Comment (bhelx) [inline src/components/PluginInput/PluginInput.tsx:40]: i deleted the unused part of the logic in App.ts so i moved it here.

File: src/components/PluginOutput/GetOutputComponent.tsx [hunk 6]
```diff
@@ -118,22 +110,20 @@ const JSONOutput: React.FC<OutputComponentProps> = ({ bytes }) => {
...
-    };
-
     const onDragStart = (e: React.DragEvent) => {
       e.dataTransfer.setData('text/uri-list', outputImageRef.current!.src);
       e.dataTransfer.dropEffect = 'move';
       e.dataTransfer.effectAllowed = 'all';
     };
+    
     return (
-      <div onDragOver={onDragPreventDefault} onDrop={onDragPreventDefault} className="output-image-component-wrapper">
```
Comment (bhelx) [inline src/components/PluginOutput/GetOutputComponent.tsx:136]: These didn't seem to be needed. We can put them back if there was a reason

File: src/components/URLInput/URLInput.tsx [hunk 7]
```diff
@@ -1,34 +1,30 @@
...
-}
+};
+
 const URLInput: React.FC<URLInputProps> = ({ onChange, url, defaultUrl }) => {
   const [urlInput, setUrlInput] = useState(url);
 
-  const handleURLChange = (e: any) => {
-    setUrlInput((prevState) => e.target.value);
+  const handleURLChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUrlInput(e.target.value);
```
Comment (bhelx) [inline src/components/URLInput/URLInput.tsx:13]: I think we can just pass a literal rather than a function?

**Comment by nilslice** [comment]
nice! lots here.. i think it would be helpful to walk through this with Joe

File: src/components/Drop-Down/HelpMenu.tsx [hunk 8]
```diff
@@ -3,17 +3,17 @@ import React, { useRef, useState } from 'react';
 const HelpMenu: React.FC = function () {
```
Comment (bhelx) [inline src/components/Drop-Down/HelpMenu.tsx:3]: Where does this get used?

Comment (nilslice) [reply src/components/Drop-Down/HelpMenu.tsx:3]: I think Joe may be waiting on design from me here.. there was a placeholder component that was a dropdown to contain the instructions to use the Playground. but i left it out of the last mockup by accident.

**Pull request merged**